plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt

```diff
@@ -142,19 +143,20 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
         }
         val supportedModules = project.getSupportedModules(supportedJavaMappings).toSet()
         val validProjectJdk = project.getSupportedJavaMappings(supportedJavaMappings).isNotEmpty()
+        val projectJdk = project.tryGetJdk()
         if (supportedModules.isEmpty() && !validProjectJdk) {
             return ValidationResult(
                 false,
                 message("codemodernizer.notification.warn.invalid_project.description.reason.invalid_jdk_versions", supportedJavaMappings.keys.joinToString()),
                 InvalidTelemetryReason(
                     CodeTransformPreValidationError.UnsupportedJavaVersion,
-                    project.tryGetJdk().toString()
+                    projectJdk?.toString() ?: ""
```

Comment:
This will change the emitted telemetry, is there a reason you changed this?

Reply:
Discussion: We own the overridden method `tryGetJdk` and can change implementation there. Look in `ModuleUtils.kt` file for overriden function `tryGetJdk`.

We could return empty string instead of null in the function that might have side effects. Your current change could be fine.